public/services/assistant_client.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { API_BASE } from '../../common/constants/llm';
+import { HttpSetup } from '../../../../src/core/public';
+
+interface Options {
+  dataSourceId?: string;
+}
+
+export class AssistantClient {
+  constructor(private http: HttpSetup) {}
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  executeAgent = (agentId: string, parameters: Record<string, any>, options?: Options) => {
```

Comment:
Will there be chance that executeAgent and executeAgentByName combined to be one API? Seems the only difference is that they pass different parameters in the `query` field

Otherwise, we need to add active method in both these 2 functions.

Reply:
this is the APIs of public(UI), they will call the corresponding server APIs, and in server side, they will end up with calling one single function.